### PR TITLE
[ES-7] Make Client class Type free

### PR DESCRIPTION
**Pull request by renanbr**
Attempt to:

- Cleanup Client methods requiring Type
- Fix ClientTest
- Copy methods from Type to Index:
    - `addDocument()`
    - `createDocument()`

Test is green `make test TEST="test/Elastica/ClientTest.php"`:

- `ClientTest::testEndpointParamsRequest()` checks against "index_total" instead of "type"

### CODE REVIEW

File: test/Elastica/ClientTest.php [hunk 1]
```diff
@@ -349,86 +308,13 @@ public function testUpdateDocuments()
...
-        $idxString = $index->getName();
-        $typeString = $type->getName();
-        $this->assertInternalType('string', $idxString);
-        $this->assertInternalType('string', $typeString);
+        $docData1 = $index->getDocument(1)->getData();
+        $docData2 = $index->getDocument(2)->getData();
 
-        // Try to delete doc with a routing value which hashes to
-        // a different shard then the id.
-        $resp = $index->getClient()->deleteIds($ids, $index, $type, 'second_routing');
```
Comment (thePanz) [inline test/Elastica/ClientTest.php]: Looks like this test was checking the "routing" too, I guess we should keep it

**Comment by ruflin** [comment]
Would  be good to get a changelog for this PR.

**Comment by thePanz** [comment]
@ruflin we have `Removed `\Elastica\Type` class, `\Elastica\Index` class must be used instead`, it should cover those changes. wdyt?

**Comment by ruflin** [comment]
SGTM. I'm happy with having an "overall" changelog entry in the end.

**Pull request merged**